packages/nextjs-mf/src/plugins/NextFederationPlugin/remove-unnecessary-shared-keys.ts

```diff
@@ -1,24 +1,32 @@
-// Utility function to remove unnecessary shared keys from the default share scope
+/**
+ * Utility function to remove unnecessary shared keys from the default share scope.
+ * It checks each key in the shared object against the default share scope.
+ * If a key is found in the default share scope, a warning is logged and the key is removed from the shared object.
+ *
+ * @param {Record<string, unknown>} shared - The shared object to be checked.
+ */
 import { DEFAULT_SHARE_SCOPE } from '../../internal';
 
-export function removeUnnecessarySharedKeys(
-  shared: Record<string, unknown>
-): void {
-  const warnings: string[] = Object.keys(shared).reduce(
-    (acc: string[], key: string) => {
-      if (DEFAULT_SHARE_SCOPE[key]) {
-        acc.push(
-          `[nextjs-mf] You are sharing ${key} from the default share scope. This is not necessary and can be removed.`
-        );
-        // Use a type assertion to inform TypeScript that 'key' can be used as an index for the 'shared' object
-        delete (shared as { [key: string]: unknown })[key];
-      }
-      return acc;
-    },
-    []
-  );
-
-  if (warnings.length > 0) {
-    console.warn('%c' + warnings.join('\n'), 'color: red');
-  }
+/**
+ * Function to remove unnecessary shared keys from the default share scope.
+ * It iterates over each key in the shared object and checks against the default share scope.
+ * If a key is found in the default share scope, a warning is logged and the key is removed from the shared object.
+ *
+ * @param {Record<string, unknown>} shared - The shared object to be checked.
+ */
+export function removeUnnecessarySharedKeys(shared: Record<string, unknown>): void {
+  Object.keys(shared).forEach((key: string) => {
+    /**
+     * If the key is found in the default share scope, log a warning and remove the key from the shared object.
+     */
+    if (DEFAULT_SHARE_SCOPE[key]) {
+      console.warn(
+        `%c[nextjs-mf] You are sharing ${key} from the default share scope. This is not necessary and can be removed.`,
+        'color: red'
+      );
+      delete (shared as { [key: string]: unknown })[key];
```

Comment:
The function 'removeUnnecessarySharedKeys' is not checking if the 'shared' object is empty or not. It would be better to add a condition to check if the object is empty and return early if it is.